Foundation/NSTask.swift

```diff
@@ -194,11 +189,69 @@ public class NSTask : NSObject {
             argv.dealloc(args.count + 1)
         }
         
-        // Lock...
+        var taskSocketPair : [Int32] = [0, 0]
+        socketpair(AF_UNIX, SOCK_STREAM, 0, &taskSocketPair)
         
-        processLaunchLock.lock()
+        var context = CFSocketContext(version: 0, info: UnsafeMutablePointer<Void>(Unmanaged.passUnretained(self).toOpaque()),
+                                               retain: runLoopSourceRetain, release: runLoopSourceRelease, copyDescription: nil)
         
-        // ...and load
+        let socket = CFSocketCreateWithNative( nil, taskSocketPair[0], CFSocketCallBackType.DataCallBack.rawValue, {
```

Comment:
also CFSocketCallBackType.DataCallBack.rawValue probably won't compile on linux and will need similar treatment as the rest of the enum types that are imported from CF